cmd/soroban-cli/src/bin/doc-gen.rs

```diff
@@ -17,17 +17,72 @@ fn doc_gen() -> std::io::Result<()> {
         .show_table_of_contents(false)
         .title("Stellar CLI Manual".to_string());
 
-    let content = clap_markdown::help_markdown_custom::<soroban_cli::Root>(&options);
+    let content = generate_markdown_with_aliases::<soroban_cli::Root>(&options);
 
     std::fs::write(out_dir.join("FULL_HELP_DOCS.md"), content)?;
 
     Ok(())
 }
 
+fn generate_markdown_with_aliases<C: clap::CommandFactory>(
+    options: &clap_markdown::MarkdownOptions,
+) -> String {
+    let command = C::command();
+    let mut markdown_content = clap_markdown::help_markdown_custom::<C>(options);
+    add_aliases_recursively(&command, &mut markdown_content, 2);
+    markdown_content
+}
+
+fn add_aliases_recursively(command: &clap::Command, content: &mut String, level: usize) {
+
+    for arg in command.get_arguments() {
+
+        let arg_name = format!("--{}", arg.get_id().as_str().to_kebab_case());
+        let mut alias_list = vec![];
+
+        let visible_aliases: Vec<String> = arg
+            .get_visible_aliases()
+            .into_iter()
+            .flatten()
+            .map(|alias| format!("--{}", alias.to_kebab_case()))
+            .collect();
+
+            alias_list.push(arg_name.clone());
+        alias_list.extend(visible_aliases);
+
+        content.push_str(&format!(
```

Comment:
This is still just adding the aliases to the end of the document. And it seems that you include the arg_name in the alias list resulting in:
```
**Argument --shell**: --shell
```

Rather you should first check if the arg has an alias and then if so find  the locations of `arg_name` in content and insert the alias after it. `--source-account, --source`. I would also suggest splitting `content` into a vec of lines and passing that in. Then in the caller join the string again. Inserting into a large string can be very expensive.  When inserting in the middle of a string, a new string is created and then first part is copied over, the new string is appended and then the end of the of the original string. So by splitting the string once you pay once for the copying and then editing each line is much cheaper. Then pay once more for the joining.

An interesting article about how Zed handles string insertions for editing files: https://zed.dev/blog/zed-decoded-rope-sumtree

Since our use case is non-interactive and a one time build it's not so important to be as efficient as possible, but I still think it's worth splitting into lines.